File: a5/ans1.pl
/** 
// Honor Pledge:
//
// I pledge that I have neither given nor 
// received any help on this assignment.
//
// gforsyth 
*/ 

/**
    Name: Gabbi Forsythe
    Date: 11/22/2019
    Class: CS441 
    Purpose: Part 1 of 3 of assignment 5 over prolog. A prolog predicate to determine what person has which aunts
*/

/** Put in all the names of the people in the family tree */
mom(barbara, monica).
mom(barbara, yvonne).
mom(barbara, sharon).
mom(monica, caroline).
mom(yvonne, emily).
mom(yvonne, courtney).
mom(sharon, maddy).
mom(sharon, samantha).
mom(emily, julie).
mom(emily, lauren). 

sister(X, Y) :- mom(Z, X), mom(Z, Y), X \= Y.
aunt(X, Y) :- sister(X, Z), mom(Z, Y).

File: a6/Ans1.ml
(**
// Honor Pledge:
//
// I pledge that I have neither given nor 
// received any help on this assignment.
//
// gforsyth 
*)

(**
    Name: Gabbi Forsythe
    Date: 12/13/2019
    Class: CS441 
    Purpose: Part 1 of 3 of assignment 6 over ML. A SML program to compute the Greatest Common Divisor
        of two positive integer values. 
*)

fun gcd (x:int, y:int)= if x = 0
                        then y
                        else if y = 0   (* if one of the numbers is equal to 0 then the other one is the GCD because math *)
                        then x
                        else if x = y (* Base Case - when the numbers are the same, that is the greatest common divisor *)
                        then x
                        else if x > y   (* Subtract big number from small number until you reach the base case *)
                        then gcd(x-y, y)
                        else gcd(x, y-x);